elements/lisk-transactions/src/4_multisignature_transaction.ts

```diff
@@ -101,10 +117,15 @@ export class MultisignatureTransaction extends BaseTransaction {
 	}
 
 	public async prepare(store: StateStorePrepare): Promise<void> {
+		const membersAddresses = extractPublicKeysFromAsset(
+			this.asset.multisignature.keysgroup,
+		).map(publicKey => ({ address: getAddressFromPublicKey(publicKey) }));
```

Comment:
[Address collision](https://github.com/LiskHQ/lisk-sdk/issues/266) is an existing problem in Lisk protocol which may lead to fetching collision accounts if we convert public keys to addresses when querying from the database. Therefore, I would prefer if we use public keys to fetch accounts instead of addresses.

Reply:
After discussion face-to-face, we concluded that it's necessary to convert public key. Converting public keys to addresses ensures that uninitialized accounts are also fetched and not overwritten with default account values. 
Nice catch @pablitovicente!